test/apps/iodemo/ucx_wrapper.cc

```diff
@@ -97,8 +101,8 @@ bool UcxContext::init()
     ucp_params.field_mask   = UCP_PARAM_FIELD_FEATURES |
                               UCP_PARAM_FIELD_REQUEST_INIT |
                               UCP_PARAM_FIELD_REQUEST_SIZE;
-    ucp_params.features     = UCP_FEATURE_TAG |
-                              UCP_FEATURE_STREAM;
+    ucp_params.features     = _use_am ? UCP_FEATURE_AM :
+                                        UCP_FEATURE_TAG | UCP_FEATURE_STREAM;
```

Comment:
can simplify , use if(_use_am) {features|=AM}

Reply:
you if want to use AM, initialize TAG and stream as well? why?

Reply:
right
anyway if/else would be more clear IMO
or at least wrap | operator with ( )